### PR TITLE
ci: prove pr title check

### PR DESCRIPTION
**Pull request by Uzlopak**
now using a regex. 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/pull-request-title.yml [hunk 1]
```diff
@@ -9,5 +9,5 @@ jobs:
     steps:
     - uses: fastify/action-pr-title@v0
       with:
-        prefixes: 'build:,chore:,ci:,docs:,feat:,fix:,perf:,refactor:,style:,test:'
+        regex: '/^(build|chore|ci|docs|feature|types|fix|perf|refactor|style|test){1}([ ]{0,1}\([^\):]*\)):/'
```
Comment (Uzlopak) [inline .github/workflows/pull-request-title.yml:12]: should it be `feat` or `feature`

I personally like "feature" more

File: .github/workflows/pull-request-title.yml [hunk 2]
```diff
@@ -9,5 +9,5 @@ jobs:
...
         github-token: ${{ github.token }}
```
Comment (Uzlopak) [inline .github/workflows/pull-request-title.yml:13]:
```suggestion
        github-token: ${{ github.token }}

```

File: .github/workflows/pull-request-title.yml [hunk 3]
```diff
@@ -9,5 +9,5 @@ jobs:
     steps:
     - uses: fastify/action-pr-title@v0
       with:
-        prefixes: 'build:,chore:,ci:,docs:,feat:,fix:,perf:,refactor:,style:,test:'
-        github-token: ${{ github.token }}
\ No newline at end of file
+        regex: '/^(build|chore|ci|docs|feature|types|fix|perf|refactor|style|test){1}([ ]{0,1}\([^\):]*\)){0,1}:/'
```
Comment (Uzlopak) [inline .github/workflows/pull-request-title.yml:12]: I changed feat to feature. wdyt

Comment (luisorbaiceta) [inline .github/workflows/pull-request-title.yml:12]:
```suggestion
        regex: '/^(build|chore|ci|docs|feature|types|fix|perf|refactor|style|test)(?:\([^\):]*\))?:\s/'
```

- The {1} after the commit message prefix group is redundant. The default behavior is to match once.
- Replace ([ ]{0,1}) with \s to match any whitespace character (spaces, tabs, etc.) after the prefix to make it more flexible.
- Modify the optional capturing group (?:\([^\):]*\))? to handle any parentheses with content within it (e.g., (some-text-here)) that might come after the prefix.
- Add \s after the optional parentheses capturing group to match any additional whitespace between the colon and the actual commit message.

Matching titles:

- "docs: Update README.md"
- "fix: Fix a bug in the login process"
- "test(some-context): Add unit tests for the API"
- "chore(build): Update build script"

Comment (Eomm) [reply .github/workflows/pull-request-title.yml]: both?

I use `feat` tbh

File: .github/workflows/pull-request-title.yml [hunk 4]
```diff
@@ -9,5 +9,5 @@ jobs:
     steps:
     - uses: fastify/action-pr-title@v0
       with:
-        prefixes: 'build:,chore:,ci:,docs:,feat:,fix:,perf:,refactor:,style:,test:'
-        github-token: ${{ github.token }}
\ No newline at end of file
+        regex: '/^(build|chore|ci|docs|feature|types|fix|perf|refactor|style|test)(?:\([^\):]*\))?:\s/'
```
Comment (Uzlopak) [inline .github/workflows/pull-request-title.yml:12]:
```suggestion
        regex: '/^(build|chore|ci|docs|feat|types|fix|perf|refactor|style|test)(?:\([^\):]*\))?:\s/'
```

File: .github/workflows/pull-request-title.yml (hunk 3 above)
Comment (Uzlopak) [reply .github/workflows/pull-request-title.yml]: i changed to feat.

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**